assets/js/components/surveys/CurrentSurvey.js

```diff
@@ -260,61 +269,40 @@ export default function CurrentSurvey() {
 				: __( 'Next', 'google-site-kit' ),
 	};
 
-	switch (
-		currentQuestion?.question_type // eslint-disable-line camelcase
-	) {
-		case 'rating':
-			surveyQuestionComponent = (
-				<SurveyQuestionRating
-					{ ...commonProps }
-					choices={ currentQuestion.question.answer_choice } // eslint-disable-line camelcase
-				/>
-			);
-			break;
-		case 'open_text':
-			surveyQuestionComponent = (
-				<SurveyQuestionOpenText
-					{ ...commonProps }
-					subtitle={ currentQuestion.subtitle }
-					placeholder={ currentQuestion.placeholder }
-				/>
-			);
-			break;
-		case 'multi_select':
-			surveyQuestionComponent = (
-				<SurveyQuestionMultiSelect
-					{ ...commonProps }
-					choices={ currentQuestion.question.answer_choice } // eslint-disable-line camelcase
-					minChoices={ currentQuestion.min_choices }
-					maxChoices={ currentQuestion.max_choices }
-				/>
-			);
-			break;
-		case 'single_select':
-			surveyQuestionComponent = (
-				<SurveyQuestionSingleSelect
-					{ ...commonProps }
-					choices={ currentQuestion.question.answer_choice } // eslint-disable-line camelcase
-				/>
-			);
-			break;
-
-		default:
-			break;
-	}
-
-	if ( ! surveyQuestionComponent ) {
-		return null;
-	}
-
 	return (
 		<Slide
 			direction="up"
 			in={ animateSurvey }
 			onExited={ handleAnimationOnExited }
 		>
 			<div className="googlesitekit-survey">
-				{ surveyQuestionComponent }
+				{ currentQuestion.question_type === 'multi_select' && (
+					<SurveyQuestionMultiSelect
+						{ ...commonProps }
+						choices={ currentQuestion.question.answer_choice }
+						minChoices={ currentQuestion.min_choices }
+						maxChoices={ currentQuestion.max_choices }
+					/>
+				) }
+				{ currentQuestion.question_type === 'open_text' && (
+					<SurveyQuestionOpenText
+						{ ...commonProps }
+						subtitle={ currentQuestion.subtitle }
+						placeholder={ currentQuestion.placeholder }
+					/>
+				) }
+				{ currentQuestion.question_type === 'rating' && (
+					<SurveyQuestionRating
+						{ ...commonProps }
+						choices={ currentQuestion.question.answer_choice }
+					/>
+				) }
+				{ currentQuestion.question_type === 'single_select' && (
+					<SurveyQuestionSingleSelect
+						{ ...commonProps }
+						choices={ currentQuestion.question.answer_choice }
+					/>
+				) }
```

Comment:
Interesting change @tofumatt  :smile: 

I like the switch statement because it's a traditional programming construct *and* it's safe in that only one thing can be returned

(actually, I've just thought about that and without eslint it's really easy to not break out of a case :yum: )

Whereas with your way of doing it, everything is inline and readable without having to scroll back up.  Something I really like in React.

This could obviously be a huge nested ternary statement but I find them really unreadable. Better to just have each condition on a new line :smile: 

This is purely just bikeshedding so don't feel that you need to reply :smile: